graylog2-web-interface/src/components/sidecars/configuration-forms/ImportsViewModal.jsx

```diff
@@ -0,0 +1,154 @@
+    CollectorConfigurationsActions.listUploads({ page: page })
+      .then(
+        (uploads) => {
+          this.setState({ uploads: uploads.uploads, totalUploads: uploads.total });
+        },
+        (error) => {
+          this.setState({ uploads: [], totalUploads: 0 });
+          UserNotification.error(`Fetching uploads failed with error: ${error}`,
+            'Could not get configuration uploads');
+        },
+      );
+  };
+
+  _onApplyButton = (selectedUpload) => {
+    this.props.onApply(selectedUpload);
+  };
+
+  _formatUpload(upload) {
+    const tooltip = <Tooltip id={`${upload.id}-status-tooltip`}>{upload.collector_id}</Tooltip>;
+
+    return (
+      <tr key={upload.id}>
+        <td>
+          <OverlayTrigger placement="top" overlay={tooltip} rootClose>
+            <span>{upload.node_id}</span>
+          </OverlayTrigger>
+        </td>
+        <td>{upload.collector_name}</td>
+        <td><Timestamp dateTime={upload.created} format={'YYYY-MM-DD HH:mm:ss z'} /></td>
+        <td>
+          <Button bsStyle="info" bsSize="xsmall" onClick={e => this._onApplyButton(upload.rendered_configuration, e)}>
+            Apply
+          </Button>
+        </td>
+      </tr>
+    );
+  }
+
+  render() {
+    if (this._isLoading()) {
+      return (
+        <BootstrapModalWrapper ref={(c) => { this.uploadsModal = c; }}>
+          <Modal.Header closeButton>
+            <Modal.Title><span>Configuration Imports</span></Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Spinner />
+          </Modal.Body>
+          <Modal.Footer>
+            <Button type="button" onClick={this.hide}>Close</Button>
+          </Modal.Footer>
+        </BootstrapModalWrapper>
+      );
+    }
+
+    const pageSize = this.PAGE_SIZE;
+    const { uploads, totalUploads } = this.state;
+    const formattedUploads = uploads.map(upload => (this._formatUpload(upload)));
+
+    if (totalUploads === 0) {
+      return (
+        <BootstrapModalWrapper ref={(c) => { this.uploadsModal = c; }}>
+          <Modal.Header closeButton>
+            <Modal.Title><span>Configuration Imports</span></Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Alert bsStyle="info">
+              <i className="fa fa-info-circle" />&nbsp;
+              There are no configuration uploads available. Please go to<br />
+              <strong>System-&gt;Collectors-&gt;Details-&gt;Import Configuration</strong><br />
+              and import your first configuration.
+            </Alert>
+          </Modal.Body>
+          <Modal.Footer>
+            <Button type="button" onClick={this.hide}>Close</Button>
+          </Modal.Footer>
+        </BootstrapModalWrapper>
+      );
+    }
+
+    return (
+      <BootstrapModalWrapper ref={(c) => { this.uploadsModal = c; }}>
+        <Modal.Header closeButton>
+          <Modal.Title><span>Configuration Imports</span></Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
```

Comment:
I think it would help to add a few words on how the user should do after they apply a configuration. At least we should mention that they need to review them and adapt them, maybe using the new variables to replace some hardcoded IDs.